### PR TITLE
DOCS-9195 new create dialogs

### PR DESCRIPTION
**Pull request by sleslie23**

### CODE REVIEW

**Review by iamerico — CHANGES_REQUESTED** [review]
Now with this new creation dialog, it is possible to create a RAML/OAS Specification or Fragment directly by importing a file or zip from option "Import from File".
This is new for API Designer, we could add a section **"Start Creating a API Specification or Fragment from an Existing One (API Designer)"**, with:
**Procedure**
1. On the Projects page of Design Center, click the Create new button and then click Import from File.
2. Specify a name to give to the imported project.
3. Click Choose file, then browse your filesystem for the .json, .raml, .yaml, .zip file that contains the project.
4. Click Import as...

**Result**
API Designer opens the new project.

File: modules/ROOT/pages/design-create-publish-api-fragment.adoc [hunk 1]
```diff
@@ -30,10 +25,14 @@ An API fragment is a RAML document that has a version and an identifier, but is
...
-. On the *Projects* page in Design Center, click *Create* and select *Create Fragment*.
-. In the *New Fragment* dialog, type a name for the project. You can change the name (but not the extension) later, if you want to.
-. Select the type of API fragment that you want to create and click *Create Fragment*.
+. On the *Projects* page in Design Center, click *Create new*.
+. Select *New Fragment*.
+. In the *New Fragment* dialog, type a name for the Fragment project. 
 +
+You can change the name (but not the extension) later, if you want to.
+
+. Select the type of API fragment that you want to create and click *Create Fragment*.
```
Comment (franciscolosardo) [inline modules/ROOT/pages/design-create-publish-api-fragment.adoc:34]: You can change the fragment type at any point. I don't know what they mean by extension though.

File: modules/ROOT/pages/design-create-publish-api-raml-editor.adoc [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
+. In the *New API Spec* dialog, name your project.
++
+You can change the name (but not the extension) later, if you want to.
 
+. Select the *I'm comfortable designing it on my own* option.
+. Choose either the RAML or OAS specification. 
++
+API Designer creates the stubs of your API specification using the version you choose.
 +
+The default is to create a RAML 1.0 file; however, you can change the file to RAML 0.8 or OAS after the project is created. Projects for creating API specifications can contain a mix of RAML and OAS files.
```
Comment (franciscolosardo) [inline modules/ROOT/pages/design-create-publish-api-raml-editor.adoc:22]: Explain RAML or OAS include multiple versions though files inside may be changed to fit any other language.

File: modules/ROOT/pages/to-create-a-mule-application-project.adoc [hunk 3]
```diff
@@ -72,11 +67,10 @@ For example, if the project contains a Database connector, the database driver f
...
 === Procedure
 
-. On the *Projects* page of Design Center, click the *Create* button and then click *Import Application*.
+. On the *Projects* page of Design Center, click the *Create new* button and then click *Import from File*.
 . Specify a name to give to the imported project.
-. Browse your filesystem for the `.jar` file that contains the project.
-// . Browse your filesystem for the `.jar` file that contains the project. Both Design Center and Studio export Mule-application projects in `.jar` files.
-. Click *Import*.
+. Click *Choose file*, then browse your filesystem for the `.jar` file that contains the project.
+. Click *Import and Create project*.
```
Comment (franciscolosardo) [inline modules/ROOT/pages/to-create-a-mule-application-project.adoc:73]: Add
"You must own an account with permissions to create mule apps. If no .jar option is shown inside import dialog, then you're missing permissions, check your account administrator"
Due to the ability of importing jar files not being showed if you're missing the rights to do so

**Review by franciscolosardo — CHANGES_REQUESTED** [review]
At least the fragment type change must be revised.
Everything else looks right! Made some comments though

File: modules/ROOT/pages/design-create-publish-api-fragment.adoc (hunk 1 above)
Comment (sleslie23) [reply modules/ROOT/pages/design-create-publish-api-fragment.adoc:34]: Changed to: "You can change the name or fragment type later, if necessary."

Not sure about "extension" either. Let me know if you think the above change doesn't reflect the behavior accurately.

File: modules/ROOT/pages/design-create-publish-api-raml-editor.adoc (hunk 2 above)
Comment (sleslie23) [reply modules/ROOT/pages/design-create-publish-api-raml-editor.adoc:22]: Added the following:

API Designer creates the stubs of your API specification using the specification type and version you choose.
+
By default API Designer creates a RAML 1.0 file. You can change the file to RAML 0.8 or OAS after the project is created. Projects for creating API specifications can contain a mix of RAML, OAS, and other types of files.

File: modules/ROOT/pages/to-create-a-mule-application-project.adoc (hunk 3 above)
Comment (sleslie23) [reply modules/ROOT/pages/to-create-a-mule-application-project.adoc:73]: Added the following:

To import a project from a `.jar` file, your account must have permissions to create Mule applications. If you do not have this permission, the option to import a jar file is not available in the *Import from File* dialog.

**Comment by sleslie23** [comment]
@iamerico Please see the new topic: Create and Publish an API Specification or Fragment from an External File

**Review by iamerico — APPROVED** [review]
LGTM

File: modules/ROOT/pages/design-create-publish-api-raml-editor.adoc [hunk 4]
```diff
@@ -1,23 +1,26 @@
...
-. Create an API-specification project that you can work in.
- .. On the *Projects* page in Design Center, click *Create* and then click *Create API Specification*.
- .. In the *New Specification* section of the dialog that is opened, name your project. You can change the name (but not the extension) later, if you want to.
- .. Select *Text editor*.
- .. Choose whether the stub specification file created in the project should be in RAML or OAS. The default is to create a RAML 1.0 file; however, you can change the file to RAML 0.8 or OAS after the project is created. In fact, projects for creating API specifications can contain a mix of RAML and OAS files.
+. On the *Projects* page in Design Center, click *Create new*.
+. Select *New API Spec*.
+. In the *New API Spec* dialog, name your project.
++
+You can change the name (but not the extension) later, if you want to.
```
Comment (franciscolosardo) [inline modules/ROOT/pages/design-create-publish-api-raml-editor.adoc:15]: I don't really recall what we're meaning with extension

File: modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc [hunk 5]
```diff
@@ -0,0 +1,51 @@
+= Create and Publish an API Specification or Fragment from an External File
+
+Use API Designer to create a new API specification or an API fragment from an existing external file. API Designer supports importing from JSON, RAML, or YML files. You can also import API specifications and fragments from an existing project.
```
Comment (franciscolosardo) [inline modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc:3]: Add zip file format

**Review by franciscolosardo — DISMISSED** [review]
Approved with some minor comments.

File: modules/ROOT/pages/design-create-publish-api-raml-editor.adoc (hunk 4 above)
Comment (sleslie23) [reply modules/ROOT/pages/design-create-publish-api-raml-editor.adoc:15]: I took out "but not the extension"

File: modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc (hunk 5 above)
Comment (sleslie23) [reply modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc:3]: Revised as:

Use API Designer to create a new API specification or an API fragment from an existing external file. API Designer supports importing from JSON, RAML, or YML files. You can also import API specifications and fragments from a ZIP file for an existing project.

File: modules/ROOT/pages/design-create-publish-api-raml-editor.adoc (hunk 4 above)
Comment (Mulesoft-Mysti) [reply modules/ROOT/pages/design-create-publish-api-raml-editor.adoc:15]: did you mean file extension (.yml)? and it is still there?

File: modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc (hunk 5 above)
Comment (Mulesoft-Mysti) [reply modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc:3]: I don't see the change, just FYI.

File: modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+== Procedure
+
+. On the *Projects* page of Design Center, click the *Create new* button and then click *Import from File*.
+. Specify a name to give to the imported project.
+. Click Choose file, then browse your filesystem for the you want to import.
```
Comment (Mulesoft-Mysti) [inline modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc:9]: Should "Choose file" be in bold?

File: modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
++
+You can create a new project from one following file types:
++
```
Comment (Mulesoft-Mysti) [inline modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc:12]: Typically, when just referring to the file type, we use JSON, RAML, YAML, and ZIP.

File: modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+API Designer opens the API specification or fragment in a new project using the text editor. API Designer uses the existing name for each file of the project. You can change these names as necessary.
+
+== Modifying Your API Specification or Fragment
+
+While you are working in the editor, you can navigate through and manipulate the content of your specification. Press F1 at any time to see a list of them. This list also displays keyboard shortcuts for most of the commands.
+
+You can also perform these tasks:
+
+Import files and API fragments into your project:: You can import files, either separately or bundled in .zip files. The files can be RAML 1.0 files, JSON files, or OpenAPI Specification (OAS) 2.0 or 3.0 files (either JSON or YAML). The files can be on your computer or you can specify a URL for them if they are located online.
```
Comment (Mulesoft-Mysti) [inline modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc:28]: ZIP (matches RAML in next sentence)

**Review by Mulesoft-Mysti — APPROVED** [review]

File: modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc (hunk 5 above)
Comment (sleslie23) [reply modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc:3]: @Mulesoft-Mysti Verified that these changes are in the branch.

File: modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc (hunk 6 above)
Comment (sleslie23) [reply modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc:9]: Added bold!

File: modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc (hunk 7 above)
Comment (sleslie23) [reply modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc:12]: Fixed.

File: modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc (hunk 8 above)
Comment (sleslie23) [reply modules/ROOT/pages/design-create-publish-api-specs-from-file.adoc:28]: Fixed!

**Review by Mulesoft-Mysti — APPROVED** [review]

**Review by iamerico — APPROVED** [review]
LGTM

**Pull request merged**